core/src/main/java/hudson/model/queue/QueueListener.java

```diff
@@ -29,6 +30,17 @@
  * @since 1.520
  */
 public abstract class QueueListener implements ExtensionPoint {
+
+    /**
+     * When a flyWeightTask is submitted to the waiting queue, but it is restricted
+     * to run on an offline node, then it gets added to the flyWeightTasks queue.
+     */
+    public void onEnterFlyWeight(FlyWeightItem fw) {}
+    /**
```

Comment:
please add spaces before the new method declarations (see the original code in this class)